lib/shared/sdk/usbboot/index.js

```diff
@@ -409,13 +443,37 @@ exports.scan = (options) => {
       debug(`Serial number index: ${serialNumberIndex}`)
 
       if (serialNumberIndex === USB_DESCRIPTOR_NULL_INDEX) {
+        // eslint-disable-next-line no-magic-numbers
+        emitProgress(10)
         return queryBlobFromCache(USBBOOT_BOOTCODE_FILE_NAME, options.readFile).then((bootcode) => {
           return writeBootCode(device, endpoint, bootcode)
         })
       }
 
       debug('Starting file server')
-      return startFileServer(device, endpoint, options.readFile)
+
+      // eslint-disable-next-line no-magic-numbers
+      emitProgress(15)
+
+      const PERCENTAGE_START = 20
+      const PERCENTAGE_TOTAL = 95
+
+      // TODO: Find a way to not hardcode these values, and instead
+      // figure out the correct number for each board on the fly.
+      // This might be possible once we implement proper device
+      // auto-discovery.
+      // eslint-disable-next-line no-magic-numbers
+      const STEPS_TOTAL = device.deviceDescriptor.idProduct === USB_PRODUCT_ID_BCM2708_BOOT ? 29 : 38
+
+      return startFileServer(device, endpoint, {
+        readFile: options.readFile,
+        progress: (step) => {
+          emitProgress((step * (PERCENTAGE_TOTAL - PERCENTAGE_START) / STEPS_TOTAL) + PERCENTAGE_START)
+        }
+      }).tap(() => {
+        // eslint-disable-next-line no-magic-numbers
+        emitProgress(100)
```

Comment:
I think we've got already got a constant for "100% progress" which we might be able to re-use here?